src/test/java/com/myhome/services/unit/UserSDJpaServiceTest.java

```diff
@@ -0,0 +1,189 @@
+package com.myhome.services.unit;
+
+import com.myhome.controllers.dto.UserDto;
+import com.myhome.controllers.dto.mapper.UserMapper;
+import com.myhome.domain.Community;
+import com.myhome.domain.CommunityAdmin;
+import com.myhome.domain.User;
+import com.myhome.repositories.UserRepository;
+import com.myhome.services.CommunityService;
+import com.myhome.services.springdatajpa.UserSDJpaService;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.test.util.ReflectionTestUtils;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.verify;
+
+class UserSDJpaServiceTest {
+
+  private String USER_ID = "test-user-id";
```

Comment:
Make it `private static final` or change name convention (uppercase with underscores is applicable to private static final fields)